### ISSUE TITLE
sometimes windows are not responsive after restoring

### ISSUE THREAD

**Issue by StT191** [Cobinja/CobiWindowList#12]
Sometimes after I restore a minimized window from the panel it doesn't respond to any input, mouse or keyboard. I need to minimize and restore the window again via clicking the icon twice in order to be able to interact with the window.

This happens only sometimes. I don't know exactly how to reproduce the issue yet.
I am using Cinnamon 3.6.7 on Linux Mint 18.3 64-bit.

**Comment by NoXPhasma** [comment]
I can confirm this and I think I know where this is coming from. For some reason, the preview window does capture the mouse and when the preview is active and you click on the app, it can happen that it doesn't give the mouse free. Just open the preview again gives the mouse free.

I'm just not sure why the preview needs to capture the mouse, if it's really needed, there should be a function which gives the mouse free in any case, when the preview closes.

**Comment by Cobinja** [comment]
Can you post the contents of $HOME/.cinnamon/glass.log and $HOME/.xsession-errors when this happens?

**Comment by NoXPhasma** [comment]
I will do that, but I can't reproduce it on will. It's unclear to me what circumstances exactly force this issue. I just know that it happens sometimes after selecting a window.

**Comment by NoXPhasma** [comment]
Ok, it just happened after I've restored a minimized window of VisualStudio Code (not that this is limited to this program). But nothing was reported in $HOME/.cinnamon/glass.log or $HOME/.xsession-errors. The .xsession-errors file is empty and here's the glass.log: https://gist.github.com/NoXPhasma/e151d0b378e0470d6b2dfde0b4df83e6

Please tell me if you need more informations on this issue. Here's my system info:
Linux Mint 18.3 x64
Cinnamon 3.6.7

**Comment by NoXPhasma** [comment]
I think I can kinda reproduce it. It seems that it is happening, if the icon is clicked in the same moment, or just before, when the preview window would open.

I've tried it now and when I click the icon in the right moment, the mouse input doesn't work till I open the window preview. It's not needed that the window in question is minimized. It's pretty hard to click it in the right time frame, but I was able to force it several times now.

**Comment by markkrj** [comment]
It happens with me also and has been like that since I started using this applet like a year ago. Don't know what exactly triggers the issue, but I think it is related to the window preview. I opened multiple firefox windows and a terminal running 'tail -f ~/.xsession-errors' and started clicking on preview of current opened firefox window and when the freeze occurred, the following error was printed: **Cjs-Message: JS LOG: pushModal: invocation of begin_modal failed**.

Edit: found this bug that suggests it is something trying to grab the cursor: https://github.com/linuxmint/cinnamon-spices-applets/issues/1610

**Comment by markkrj** [comment]
Remembered another issue (maybe same bug?) that happens with me. I use guake (quake style terminal) and if I press the hotkey to open it while showing a preview (when hovering an icon or window preview) it does not open. And if I'm working in my terminal and hover an icon, the preview appears grabbing my mouse and making my guake terminal disappear. Maybe that is better to reproduce and fix. Is it necessary to grab the pointer for the window previews to work properly?

**Comment by barreira** [comment]
One more user here reporting the same bug. From my experience, the programs that are running have nothing to do with the bug itself (i.e. it happens with a lot of programs).

**Comment by Cobinja** [comment]
I'm investigating.
It seems to me that it happens when clicking an app button right before the preview is opened automatically.
Can anyone confirm?

**Comment by NoXPhasma** [comment]
Right, that's what I found out as well, see https://github.com/Cobinja/CobiWindowList/issues/12#issuecomment-370012027.

**Comment by KevinNadaud** [comment]
I success to reproduce the bug using @NoXPhasma advise: 
You need to click before the animation, so you have to be sufficiently quick but it works almost each time.
